trunk/SUAVE/Methods/Aerodynamics/AVL/AVL_Callable.py

```diff
@@ -0,0 +1,384 @@
+                        #for d in self.default_case.stability_and_control.control_deflections:
+                                #if d.tag not in case.stability_and_control.control_deflections.keys():
+                                        #case.append_control_deflection(d.tag,d.deflection)
+
+        self.settings.run_cases.num_cases += 1
+        case.index = self.settings.run_cases.num_cases # index starting at 1, consistent with AVL convention
+
+        self.settings.run_cases.cases.append(case)
+
+        return
+
+
+    def append_cases(self,cases):
+        # assert database type
+        if not isinstance(cases,Data):
+            raise Component_Exception, 'input cases must be of type Data()'
+
+        for case in cases.cases:
+            self.append_case(case)
+
+        return
+
+
+    def __call__(self,cases):
+        """ process vehicle to setup geometry, condititon and configuration
+
+            Inputs:
+                conditions - DataDict() of aerodynamic conditions
+
+            Outputs:
+                CL - array of lift coefficients, same size as alpha
+                CD - array of drag coefficients, same size as alpha
+
+            Assumptions:
+                linear intperolation surrogate model on Mach, Angle of Attack
+                    and Reynolds number
+                locations outside the surrogate's table are held to nearest data
+                no changes to initial geometry or configuration
+
+        """
+        if cases is not None and len(cases) > 0:
+            self.append_cases(cases)
+
+        write_geometry(self)
+        write_run_cases(self)
+        write_input_deck(self)
+
+        results = run_analysis(self)
+
+        # unpack filenames
+        files_path        = self.settings.filenames.run_folder
+        results_files     = [case.result_filename for case in self.settings.run_cases.cases]
+        geometry_filename = self.settings.filenames.features
+        cases_filename    = self.settings.filenames.cases
+        deck_filename     = self.settings.filenames.input_deck
+        
+        if not self.keep_files:
+            from purge_directory import purge_directory
+            purge_directory(self.settings.filenames.run_folder,purge_subdirectories=False)
+
+        return results
+
+
+
+
+
+##############################
+## Methods used in __call__ ##
+##############################
+
+def write_geometry(self):
+    # imports
+    from .write_geometry import make_header_text,translate_avl_wing,make_surface_text,translate_avl_body,make_body_text
+
+    # unpack inputs
+    aircraft      = self.features
+    files_path    = self.settings.filenames.run_folder
+    geometry_file = self.settings.filenames.features
+    geometry_path = os.path.abspath(os.path.join(files_path,geometry_file))
+
+    # Open the geometry file after purging if it already exists
+    purge_files([geometry_path])
+
+    geometry = open(geometry_path,'w')
+
+    try:
```

Comment:
python closes the file if it exits on an exception.  this structure shouldn't be needed